src/main/java/uk/nhs/hee/tis/trainee/sync/facade/ProgrammeMembershipEnricherFacade.java

```diff
@@ -85,9 +92,46 @@ public void enrich(ProgrammeMembership programmeMembership) {
       // TODO: validate the aggregated data to ensure we have a "complete" PM?
       ConditionsOfJoining conditionsOfJoining = getConditionsOfJoining(programmeMembership);
 
-      AggregateProgrammeMembershipDto aggregateProgrammeMembership =
-          aggregateMapper.toAggregateProgrammeMembershipDto(programmeMembership, programme,
-              aggregatedCurriculumMemberships, conditionsOfJoining);
+      return aggregateMapper.toAggregateProgrammeMembershipDto(programmeMembership, programme,
+          aggregatedCurriculumMemberships, conditionsOfJoining);
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Broadcast a conditions of joining sync event.
+   *
+   * @param programmeMembership The programme membership to which the Conditions of Joining
+   *                            belongs.
+   */
+  public void broadcastCoj(ProgrammeMembership programmeMembership) {
+    AggregateProgrammeMembershipDto aggregatePmDto
+        = buildAggregateProgrammeMembershipDto(programmeMembership);
+
+    if (aggregatePmDto != null) {
+      Record programmeMembershipEventRecord = eventMapper.toRecord(aggregatePmDto);
+      programmeMembershipEventRecord.setOperation(LOAD);
+      programmeMembershipEventRecord.setTable(ConditionsOfJoining.ENTITY_NAME);
```

Comment:
Not a huge fan of this, but it allows it to use the correct eventNotificationProperties . Possibly the `eventNotificationProperties` should be merged with the `BroadcastRouting` in some way